File: Day-6/day6.py
file = open("Day-6/input.txt", "r")

dataStream = file.readline().strip()

def messageDetection(size, data):
    left = 0
    right = size

    while (right < len(data) + 1):

        curr = data[left:right]

        if len(curr) == len(set(curr)):
            break;

        left += 1
        right += 1

    return right

print("Solution to part 1: ", messageDetection(4, dataStream))
print("Solution to part 2: ", messageDetection(14, dataStream))

File: Day-6/day6.hs
import Data.List

main = do
    fileData <- readFile "input.txt"
    let partA = messageDetection fileData 4 0
    let partB = messageDetection fileData 14 0
    print $ "Solution to part 1: " ++ show partA
    print $ "Solution to part 2: " ++ show partB


messageDetection :: String -> Int -> Int -> Int 
messageDetection dataStream size index | (length . nub) (take size dataStream) == size = index + size
                                       | otherwise = messageDetection (tail dataStream) size (index + 1)
